File: PreUtil/src/output/c/hard/PalindromePartitioningIii.c
/**https://leetcode.com/problems/palindrome-partitioning-iii/ */
//You are given a string s containing lowercase letters and an integer k. You need to :
//First, change some characters of s to other lowercase English letters.
//Then divide s into k non-empty disjoint substrings such that each substring is a palindrome.
//Return the minimal number of characters that you need to change to divide the string.


int palindromePartition(char * s, int k){
    int i,j;
    int* arr = (int*)malloc(sizeof(int)*1000);
    for(i=0;i<1000;i++){
        arr[i] = 0;
    }
    int count = 0;
    int temp = 0;
    for(i=0;i<strlen(s);i++){
        if(s[i]=='-'){
            arr[count] = temp;
            temp = 0;
            count++;
        }
        else{
            temp = temp*10 + (s[i]-'0');
        }
    }
    arr[count] = temp;
    count++;
    int* arr2 = (int*)malloc(sizeof(int)*1000);
    for(i=0;i<1000;i++){
        arr2[i] = 0;
    }
    int count2 = 0;
    int temp2 = 0;
    for(i=0;i<strlen(s);i++){
        if(s[i]=='-'){
            arr2[count2] = temp2;
            temp2 = 0;
            count2++;
        }
        else{
            temp2 = temp2*10 + (s[i]-'0');
        }
    }
    arr2[count2] = temp2;
    count2++;
    int* arr3 = (int*)malloc(sizeof(int)*1000);
    for(i=0;i<1000;i++){
        arr3[i] = 0;
    }
    int count3 = 0;
    int temp3 = 0;
    for(i=0;i<strlen(s);i++){
        if(s[i]=='-'){
            arr3[count3] = temp3;
            temp3 = 0;
            count3++;
        }
        else{
            temp3 = temp3*10 + (s[i]-'0');
        }
    }
    arr3[count3] = temp3;
    count3++;
    int* arr4 = (int*)malloc(sizeof(int)*1000);
    for(i=0;i<1000;i++){
        arr4[i] = 0;
    }
    int count4 = 0;
    int temp4 = 0;
    for(i=0;i<strlen(s);i++){
        if(s[i]=='-'){
            arr4[count4] = temp4;
            temp4 = 0;
            count4++;
        }
        else{
            temp4 = temp4*10 + (s[i]-'0 ');

}

    

File: PreUtil/src/output/python/hard/PalindromePartitioningIii.py
#https://leetcode.com/problems/palindrome-partitioning-iii/
#You are given a string s containing lowercase letters and an integer k. You need to :
#First, change some characters of s to other lowercase English letters.
#Then divide s into k non-empty disjoint substrings such that each substring is a palindrome.
#Return the minimal number of characters that you need to change to divide the string.


class Solution(object):
    def palindromePartition(self, s, k):
        """
        :type s: str
        :type k: int
        :rtype: int
        """      
        def get_depth(s):
            res = 0
            for c in s:
                if c == '-':
                    res += 1
                else:
                    break
            return res
        def get_val(s):
            res = 0
            for c in s:
                if c == '-':
                    break
                else:
                    res = res * 10 + int(c)
            return res
        def get_node(s):
            return TreeNode(get_val(s))
        def get_children(s):
            res = []
            i = 0
            while i < len(s):
                if s[i] == '-':
                    j = i
                    while j < len(s) and s[j] == '-':
                        j += 1
                    k = j
                    while k < len(s) and s[k] != '-':
                        k += 1
                    res.append(s[j:k])
                    i = k
                else:
                    i += 1
            return res
        def get_tree(s):
            if not s:
                return None
            root = get_node(s)
            children = get_children(s)
            if not children:
                return root
            root.left = get_tree(children[0])
            if len(children) > 1:
                root.right = get_tree(children[1])
            return root
        return get_tree(traversal)

        
        
        
    